framework/plugin/settings/checkbox.php

```diff
@@ -38,7 +38,7 @@ function render_setting_field_checkbox($config) {
     <?php elseif ($type === 'switch'): ?>
         <div class="tangible-card">
             <label class="tangible-feature-switch" role="switch" aria-checked="<?php echo $checked ? 'true' : 'false'; ?>" tabindex="0" 
-            onclick="this.querySelector('input[type=checkbox]').click();" 
+            onclick="this.querySelector('input[type=checkbox]').checked = ! this.querySelector('input[type=checkbox]').checked;"
```

Comment:
```suggestion
            onclick="this.querySelector('input[type=checkbox]').checked ^= 1"
```

There's a shorthand syntax to toggle the checked state using the bitwise XOR operator `^`.

```js
input.checked ^= 1
```

It works like:

| a | b | a ^ b |
|---|---|---|
| 1 | 1 | 0 |
| 0 | 1 | 1 |

Reply:
Thanks, I didn't know about it!